### PR TITLE
Cart refactor

### PR DESCRIPTION
**Pull request by EasileleM**
**WORK IN PROGRESS**
- deleted redundant code
- created some basic redux stuff to implement cart functionality

### CODE REVIEW

File: client/src/components/CartWindow/Items/Items.jsx [hunk 1]
```diff
@@ -4,14 +4,15 @@ import './Items.scss';
...
-  const items = props.data.map((item) => {
-    return <Item
-            key={item.item._id}
-            data={item}
-            itemAmountChange={(id, amount) => props.itemAmountChange(id, amount)}
-            deleteItem={(id) => props.deleteItem(id)}
-          />
-  });
+  const items = []
+  // props.data.map((item) => {
```
Comment (miffn3) [inline client/src/components/CartWindow/Items/Items.jsx]: comments in pr

File: client/src/components/CartWindow/OrderBlock/OrderBlock.jsx [hunk 2]
```diff
@@ -3,9 +3,10 @@ import React from 'react';
 import './OrderBlock.scss';
 
 export function OrderBlock(props) {
-  const orderPrice = props.data.reduce((price, item) => {
-    return price + item.item.price * item.amount;
-  }, 0);
+  const orderPrice = 100;
+  // props.data.reduce((price, item) => {
```
Comment (miffn3) [inline client/src/components/CartWindow/OrderBlock/OrderBlock.jsx]: same

File: client/src/utils/cart/getCartItems.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+// export function getCartItems() {
+//   return dispatch => {
+//     dispatch(getCartItemsBegins());
+//     return fetch("/products")
```
Comment (miffn3) [inline client/src/utils/cart/getCartItems.js]: same

File: client/package.json [hunk 4]
```diff
@@ -19,6 +19,7 @@
     "react-scripts": "3.2.0",
     "react-toastify": "^5.4.0",
     "redux": "^4.0.4",
+    "redux-devtools-extension": "^2.13.8",
     "redux-thunk": "^2.3.0"
   },
   "scripts": {
```
Comment (fastndead) [inline client/package.json]: isn't "redux-devtools-extension" a dev dependency?

File: client/src/reducers/cartReducer.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import { CART_ACTIONS } from '../actions/types';
+import addItem from '../utils/cart/addItem';
+import removeItem from '../utils/cart/removeItem';
+import changeItem from '../utils/cart/changeItem';
+
+const initialState = {
+  size: 0,
+  opened: false,
+  loading: false,
+  items: []
```
Comment (fastndead) [inline client/src/reducers/cartReducer.js]: You put all these properties in our common store. This is no good, i feel like, wouldn't it be better to create property 'cart' to store all those properties in it?

Comment (fastndead) [reply client/src/reducers/cartReducer.js]: or at least rename it to 'cartOpened'

File: client/src/components/CartWindow/CloseButton/CloseButton.scss [hunk 6]
```diff
@@ -3,11 +3,9 @@
     z-index: 5;
     display: flex;
     flex-direction: row-reverse;
-    border-top-right-radius: 4px;
-    border-top-left-radius: 4px;
     width: 100%;
-    background: #ff5912;
-    box-shadow: 0px 2px 17px 0px rgba(0,0,0,0.75);
+    background: #ffffff;
```
Comment (KirilushkinAleksandr) [inline client/src/components/CartWindow/CloseButton/CloseButton.scss]: Our project has variables for colors, they have to be used

File: client/src/components/CartWindow/CloseButton/CloseButton.scss [hunk 7]
```diff
@@ -16,15 +14,15 @@
     background: transparent;
     outline: 0;
     cursor: pointer;
-    color: #3c4751;
+    color: #ff5912;
```
Comment (KirilushkinAleksandr) [inline client/src/components/CartWindow/CloseButton/CloseButton.scss]: Our project has variables for colors, they has to be used

File: client/src/components/CartWindow/Items/Item/Item.scss [hunk 8]
```diff
@@ -31,7 +31,8 @@
     &-wrapper {
       @include font-primary-semi-bold-sm;
       margin: 0;
-      margin-right: 30px;
+      margin-right: 25px;
+      width: 50px;
       color: #3c4751;
```
Comment (KirilushkinAleksandr) [inline client/src/components/CartWindow/Items/Item/Item.scss]: Our project has variables for colors, they has to be used

File: client/src/components/CartWindow/Items/Items.scss [hunk 9]
```diff
@@ -17,7 +17,7 @@
      
     &::-webkit-scrollbar-thumb {
       border-radius: 4px;
-      background: #495662; 
+      background: #ff5912;
```
Comment (KirilushkinAleksandr) [inline client/src/components/CartWindow/Items/Items.scss]: Our project has variables for colors, they has to be used

File: client/src/components/CartWindow/OrderBlock/OrderBlock.scss [hunk 10]
```diff
@@ -5,14 +5,12 @@
   display: flex;
   justify-content: space-between;
   align-items: center;
-  height: 60px;
-  border-bottom-left-radius: 4px;
-  border-bottom-right-radius: 4px;
-  background: #ff5912;
-  box-shadow: 0px 2px 17px 0px rgba(0,0,0,0.75);
+  height: 100px;
+  background: #ffffff;
```
Comment (KirilushkinAleksandr) [inline client/src/components/CartWindow/OrderBlock/OrderBlock.scss]: Our project has variables for colors, they has to be used

File: client/src/components/CartWindow/OrderBlock/OrderBlock.scss [hunk 11]
```diff
@@ -5,14 +5,12 @@
...
+  box-shadow: 0px 2px 17px 0px rgba(0,0,0,0.15);
 
   &-price{
-    color: #ffffff;
+    color: #ff5912;
```
Comment (KirilushkinAleksandr) [inline client/src/components/CartWindow/OrderBlock/OrderBlock.scss]: Our project has variables for colors, they has to be used

File: client/src/components/CartWindow/OrderBlock/OrderBlock.scss [hunk 12]
```diff
@@ -22,15 +20,17 @@
...
 
   &-button {
-    @include font-primary-bold-md;
+    @include font-primary-regular-md;
     transition: 200ms ease;
     margin-right: 40px;
     border-radius: 20px;
-    background-color: #ffffff;
+    border: 0;
+    background-color: #ff5912;
```
Comment (KirilushkinAleksandr) [inline client/src/components/CartWindow/OrderBlock/OrderBlock.scss]: Our project has variables for colors, they has to be used

File: client/src/components/CartWindow/OrderBlock/OrderBlock.scss [hunk 13]
```diff
@@ -22,15 +20,17 @@
...
     cursor: pointer;
     outline: 0;
-    color: #3c4751;
-    line-height: 20px;
+    color: #ffffff;
```
Comment (KirilushkinAleksandr) [inline client/src/components/CartWindow/OrderBlock/OrderBlock.scss]: Our project has variables for colors, they has to be used

File: client/src/components/Header/index.js [hunk 14]
```diff
@@ -9,7 +9,7 @@ export class Header extends React.Component {
     return (
       <header className='header'>
         <Contacts />
-        <Navigation cartSize={this.props.cartSize} handleOnClickOpenCart={() => this.props.handleOnClickOpenCart()}/>
+        <Navigation />
```
Comment (KirilushkinAleksandr) [inline client/src/components/Header/index.js]: Unnecessary spacing after Navigation

File: client/src/components/Header/navigation.js [hunk 15]
```diff
@@ -51,7 +51,7 @@ class Navigation extends Component {
                   </label>
                 </form>
                 <User />
-                <ShopCart cartSize={this.props.cartSize} handleOnClickOpenCart={() => this.props.handleOnClickOpenCart()} />
+                <ShopCart />
```
Comment (KirilushkinAleksandr) [inline client/src/components/Header/navigation.js]: Unnecessary spacing after ShopCart

File: client/src/components/Product/index.js [hunk 16]
```diff
@@ -49,7 +49,7 @@ class Product extends Component {
                 <div className="product">
                     <ProductImages
                         images={this.state.item.images}/>
-                    <ProductInfo addToCard={(item, size, color, amount) => this.props.addToCard(item, size, color, amount)} item={this.state.item}/>
+                    <ProductInfo item={this.state.item}/>
```
Comment (KirilushkinAleksandr) [inline client/src/components/Product/index.js]: Unnecessary spacing

**Pull request merged**